src/app/archive/lib/processor.ml

```diff
@@ -623,14 +658,24 @@ module Zkapp_permissions = struct
   let table_name = "zkapp_permissions"
 
   let add_if_doesn't_exist (module Conn : CONNECTION) (perms : Permissions.t) =
+    let open Deferred.Result.Let_syntax in
+    let version = snd perms.set_verification_key in
+    let%bind protocol_version_id =
+      Protocol_versions.add_if_doesn't_exist
+        (module Conn)
+        ~transaction:(Protocol_version.transaction version)
+        ~network:(Protocol_version.network version)
+        ~patch:(Protocol_version.patch version)
+    in
     let value =
       { edit_state = perms.edit_state
       ; send = perms.send
       ; receive = perms.receive
       ; access = perms.access
       ; set_delegate = perms.set_delegate
       ; set_permissions = perms.set_permissions
-      ; set_verification_key = perms.set_verification_key
+      ; set_verification_key = fst perms.set_verification_key
+      ; set_verification_key_version = protocol_version_id
```

Comment:
should this value come from `perms.set_verification_key`

Reply:
Hmm, not sure what do you mean here. This function tries to add `Permission` to the database, and first I add protocol_version and then get its id and use that id in this field here.

Reply:
Sorry, I didn't see the code above this one.

But if `set_verification_key` contains only the transaction version, then you can use the `snd` of that directly.

Reply:
Yeah. I am not sure whether we want the entire protocol version in permission or not. If we change this in sql then we should also change it in ocaml for the same reason.